### ISSUE TITLE
@auth not working with custom cypher from 1.1.0

### ISSUE THREAD

**Issue by MarianAlexandruAlecu** [neo4j/graphql#326]
**Describe the bug**
```
@neo4j/graphql:execute Neo4jError: Invalid input 'WHERE': expected
  "CALL"
  "CREATE"
  "DELETE"
  "DETACH"
  "FOREACH"
  "LOAD"
  "MATCH"
  "MERGE"
  "OPTIONAL"
  "REMOVE"
  "RETURN"
  "SET"
  "UNION"
  "UNWIND"
  "USE"
  "WITH"
  <EOF> (line 6, column 1 (offset: 177))
"WHERE apoc.util.validatePredicate(NOT(EXISTS(this.id) AND this.id = $this_email_auth_allow0_id), "@neo4j/graphql/FORBIDDEN", [0])"
 ^
: 
    at captureStacktrace (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/result.js:239:17)
    at new Result (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/result.js:59:23)
    at newCompletedResult (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/transaction.js:372:12)
    at Object.run (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/transaction.js:226:20)
    at Transaction.run (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/transaction.js:98:34)
    at /Users/alex/Documents/Code/teenera/backend/api/node_modules/@neo4j/graphql/src/utils/execute.ts:84:16
    at TransactionExecutor._safeExecuteTransactionWork (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:92:26)
    at TransactionExecutor._executeTransactionInsidePromise (/Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:83:34)
    at /Users/alex/Documents/Code/teenera/backend/api/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:40:19
    at new Promise (<anonymous>) {
  constructor: [Function: Neo4jError],
  code: 'Neo.ClientError.Statement.SyntaxError'
```

**Type definitions**

```graphql
type Query {
  getSelf: [User]!
    @cypher(
      statement: """
      MATCH (user:User { id: $auth.jwt.sub })
      RETURN user;
      """
    )
}
type User {
  id: ID! @id
  name: String!
  # @auth is breaking the resolver
  email: String! @auth(rules: [{ operations: [READ], allow: { id: "$jwt.sub" } }])
  password: String! @private
}
```

**To Reproduce**

```graphql
query GetSelf {
  getSelf {
    id
    name
    email
  }
}
```

The resulted cypher is
```
@neo4j/graphql:execute About to execute Cypher:
Cypher:

                WITH apoc.cypher.runFirstColumn("MATCH (user:User { id: $auth.jwt.sub })
RETURN user;", {auth: $auth}, true) as x
                UNWIND x as this
            
WHERE apoc.util.validatePredicate(NOT(EXISTS(this.id) AND this.id = $this_email_auth_allow0_id), "@neo4j/graphql/FORBIDDEN", [0])
RETURN this { .id, .name, .email } AS this
Params:
{
  "auth": {
    "isAuthenticated": true,
    "roles": "user",
    "jwt": {
      "sub": "b6b5beb8-c1da-43f8-b0e6-bf321ec8a3d5",
      "roles": "user",
      "iat": 1626418359
    }
  },
  "this_email_auth_allow0_id": "b6b5beb8-c1da-43f8-b0e6-bf321ec8a3d5"
}
```

**Expected behavior**

Custom `@cypher` resolver should work with `@auth` (as it used to before 1.1.0). If I remove `@auth` from `email` field, it's working and the resulted query is:

```
@neo4j/graphql:execute About to execute Cypher:
Cypher:

                WITH apoc.cypher.runFirstColumn("MATCH (user:User { id: $auth.jwt.sub })
RETURN user;", {auth: $auth}, true) as x
                UNWIND x as this
            
RETURN this { .id, .name, .email } AS this
Params:
{
  "auth": {
    "isAuthenticated": true,
    "roles": "user",
    "jwt": {
      "sub": "b6b5beb8-c1da-43f8-b0e6-bf321ec8a3d5",
      "roles": "user",
      "iat": 1626418359
    }
  }
}
```

**System (please complete the following information):**

-   OS: macOS
-   Version: @neo4j/graphql@1.1.0 (tested up to 1.2.1)
-   Node.js version: 14.17.0

**Comment by MarianAlexandruAlecu** [comment]
I have some fields on `type User` that I want to be accessible only to self - am I missing something with my `@auth` implementation?

**Comment by darrellwarde** [comment]
Hey @MarianAlexandruAlecu, this fix has now been released in 1.2.2 (https://github.com/neo4j/graphql/releases/tag/%40neo4j%2Fgraphql%401.2.2). Have a great weekend!
